Khan/khan-linter
# Add babel-eslint-plugin

We'd like to use the no-invalid-this rule to catch incorrect uses
of this in our code.  Unfortunately, the built-in version of this
rule doesn't work with class properties which we use throughout
our code.  This PR adds babel-eslint-plugin which contains a rule
called babel/no-invalid-this which does the same check but works
with class properties.

Plugin details: https://github.com/babel/eslint-plugin-babel

Test Plan:
- enable babel/no-invalid-this in webapp's .babelrc
- run ka-lint javascript/video-package/video-transcript-view.jsx
- see that it picks up the error that was fixed in https://phabricator.khanacademy.org/D56166

Reviewers: #fe-infra
